### PR TITLE
Kernel: Fix i8042 scan code setting sequence for bare metal hardware

### PR DESCRIPTION
**Pull request by supercomputer7**
cc @linusg

### CODE REVIEW

**Comment by linusg** [comment]
Tested on my machine, works!

File: Kernel/Arch/x86/ISABus/I8042Controller.cpp [hunk 1]
```diff
@@ -111,16 +111,21 @@ UNMAP_AFTER_INIT ErrorOr<void> I8042Controller::detect_devices()
         configuration &= ~I8042ConfigurationFlag::FirstPS2PortInterrupt;
         configuration &= ~I8042ConfigurationFlag::SecondPS2PortInterrupt;
 
+        // Note: Drain output buffer to ensure we are in a known state before setting
+        // the scan code.
+        TRY(drain_output_buffer());
```
Comment (8infy) [inline Kernel/Arch/x86/ISABus/I8042Controller.cpp:116]: I don't think this is safe if user is mashing keyboard keys/moving the mouse (as people sometimes do at PC startup). Maybe a better approach would be to consume any garbage as part of the expected output? Even if they aren't mashing the keys there's still a race window here.

**Comment by supercomputer7** [comment]
What I think is the best to do with the problem of setting scan code 2 is to gate that piece of code with a kernel commandline option. I will look into this today.

**Comment by supercomputer7** [comment]
Let's try this again.

**Comment by supercomputer7** [comment]
All of these patches work both on QEMU and on my ICH7 test machine, so @linusg please test them and if they are OK, we should merge these :)

**Comment by supercomputer7** [comment]
This had conflicts so I rebased it now again.

**Review by linusg — CHANGES_REQUESTED** [review]

**Pull request merged**